### ISSUE TITLE
[Hold for #16923] [$1000] Multiple sticky reaction menu opens 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22189]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open website and Click any chat
2. Hover over any message and click "Mark as unread" and hover up the mouse to another message above fast.
## Expected Result: 
There shouldn't be multiple reaction boxes
## Actual Result: 
Multiple reaction menu opens up

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.36-2
**Reproducible in staging?:**  y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[screen-capture (4).webm](https://github.com/Expensify/App/assets/43996225/df1886ec-84b6-4aca-9534-340725a62124)

![Screenshot from 2023-07-03 16-10-28](https://github.com/Expensify/App/assets/43996225/f573b3be-19b9-4a31-ac8b-eeff6fcd8d8a)

https://github.com/Expensify/App/assets/43996225/da19eabb-b1a6-4d67-8796-bf96c533b754

**Expensify/Expensify Issue URL:**
**Issue reported by:** @muxriddinmuqimov77
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688382583953859

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0197b945e77851e98e</li>
        <li>Upwork Job ID: 1678382570860851200</li>
        <li>Last Price Increase: 2023-07-10</li>
    </ul>
</details>

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Multiple sticky reaction menu opens

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/e23cd16d8cd9818c6477e2ba9e5a4cedd2d1f3dd/src/pages/home/report/ContextMenu/ContextMenuActions.js#L224-L230
1. `Report.markCommentAsUnread(reportID, reportAction.created)` sets `optimisticData` in `Onyx`;
2. JS Thread gets locked for some time;
3. If you move the mouse out while the JS Thread is busy, the `context menu` will not get dismissed.
### What changes do you think we should make in order to solve the problem?
Instead of using:
```js
Report.markCommentAsUnread(reportID, reportAction.created);
```
We may use:
```js
setTimeout(() => {
    Report.markCommentAsUnread(reportID, reportAction.created);
}, 1);
```
1ms adds basically no delay and no matter how fast we move the mouse, the `context menu` will get dismissed.

**Comment by NicMendonca** [comment]
I cannot reproduce this

![2023-07-05_15-36-56 (1)](https://github.com/Expensify/App/assets/24947345/0550f742-7381-4890-bf37-4664b6577159)

**Comment by DanutGavrus** [comment]
@NicMendonca Could you try clicking and directly moving your mouse out? If you wait after clicking it won't happen for you.

**Comment by NicMendonca** [comment]
To be fair, I feel as though I moved as "quickly" as possible 😅

**Comment by NicMendonca** [comment]
Getting a gut check here: https://expensify.slack.com/archives/C01SKUP7QR0/p1688644216366339

**Comment by DanutGavrus** [comment]
@NicMendonca Yes, you moved very quick indeed, so the movement speed may not be the problem.  

It seems like there is some time between clicking and starting to move the mouse instead of "click & move", which may be the problem.

**Comment by muxriddinmuqimov77** [comment]
@NicMendonca  This video might help you understand better.
[screen-capture (16).webm](https://github.com/Expensify/App/assets/101332597/eedccc23-5582-4b3b-a291-601caf1b3032)

**Comment by NicMendonca** [comment]
@dylanexpensify was able to reproduce, so moving this along

**Comment by s77rt** [comment]
Looks like a dupe of https://github.com/Expensify/App/issues/16923

**Comment by MahmudjonToraqulov** [comment]
> Looks like a dupe of #16923

Hi @s77rt. I'm not sure exactly but it looks a bit different for me.

**Comment by s77rt** [comment]
@MahmudjonToraqulov The reproduction steps are a bit different but the root cause (missing `mouseleave` event) seems to be the same.

**Comment by DanutGavrus** [comment]
@s77rt
1. If the [root cause](https://github.com/Expensify/App/issues/16923#issuecomment-1540926607) from that issue
> [...] if the element changes its position and the cursor remains static (due to a transition or some other element popping up and causing the original element to shift a little), the element would be stuck in the hovered state.  

is correct, then the problem there is that the *elemnt moves, thus the onMouseLeave does not trigger*, and if the [root cause](https://github.com/Expensify/App/issues/22189#issuecomment-1620723761) from this issue:
> 3. If you move the mouse out while the JS Thread is busy, the context menu will not get dismissed;  

is correct, then the *mouse actually moves out from the component and onMouseLeave should trigger*;  

2. This issue is reproducible only if clicking on the `Mark as unread` which has `Onyx processing`, no other action causes this issue, which means that this root cause may be correct and different;
3. I've added a log in our `Hoverable` component inside `onBlur`, it does not trigger until click somewhere out of the component, which means that any fix which includes `onBlur` would not work here as we may still open multiple sticky menus until clicking somewhere out.

Based on these I'm inclined to think that these two are different issues, but please let me know your analysis, maybe I've got something wrong.

**Comment by s77rt** [comment]
@DanutGavrus I still think there is a potential that the solution of the first issue will also fix this one.

BTW, we do not use `setTimeout` unless it's absolutely necessary (the purpose is to actual to delay some process). Using it in this situation looks hacky.

**Comment by DanutGavrus** [comment]
@s77rt
> I still think there is a potential that the solution of the first issue will also fix this one.

Yes, you might be right, there are quite some Proposals with different approaches there.

> [...] looks hacky.  

Indeed, it would be more like a workaround. Hopefully, a better solution will be found.

**Comment by NicMendonca** [comment]
holdin'

**Comment by NicMendonca** [comment]
https://github.com/Expensify/App/issues/16923 is off hold, but they are saying its no longer reproducible

**Comment by NicMendonca** [comment]
@muxriddinmuqimov77 @kavimuru - since I was never able to reproduce this, can you all try?

**Comment by muxriddinmuqimov77** [comment]
> @muxriddinmuqimov77 @kavimuru - since I was never able to reproduce this, can you all try?

@NicMendonca Yes, It's still reproducible for me. 

[screen-capture (77).webm](https://github.com/Expensify/App/assets/101332597/b084f2b3-36cc-48cf-8a34-cc64e17cd448)

**Comment by NicMendonca** [comment]
Okay! @s77rt any thoughts on the above proposals/ comments so far?

**Comment by s77rt** [comment]
@NicMendonca This issue is still on hold for #16923. Nevertheless currently there is only [one proposal](https://github.com/Expensify/App/issues/22189#issuecomment-1620723761) which uses `setTimeout`. That is a workaround and it's strictly discouraged.

For now, we can still keep holding and retest after #16923 is resolved.

**Comment by NicMendonca** [comment]
https://github.com/Expensify/App/issues/16923 is not yet resolved

**Comment by NicMendonca** [comment]
same update ^^

**Comment by s77rt** [comment]
No longer on hold and issue does not seem reproducible now

**Comment by s77rt** [comment]
@NicMendonca I think we can close this. Does not seem reproducible anymore
